os/drivers/serial/serial.c

```diff
@@ -125,6 +135,56 @@ static const struct file_operations g_serialops = {
  * Private Functions
  ************************************************************************************/
 
+static void add_char_to_log_net(char ch)
+{
+	if (log_net_head == NULL) { /* linked list is empty */
+		/* no chunks are present so we allocate new chunk */
+		struct log_net_chunk_s *node = malloc (sizeof (struct log_net_chunk_s));
```

Comment:
We should use `kmm_malloc` instead of `malloc` from kernel side for loadable apps.
Please check this.